### PR TITLE
Use Sync Management components in Export use case

### PR DESCRIPTION
**Pull request by yeldah3**
https://www.loom.com/share/bb2fe66174d94d6a93daafe5764a9469

Before:
Screenshot 2024-05-24 at 4 35 55 PM

After:
Screenshot 2024-05-24 at 4 35 22 PM

### CODE REVIEW

**Comment by yeldah3** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sutrolabs/census-embedded-demo/43?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#43** <a href="https://app.graphite.dev/github/pr/sutrolabs/census-embedded-demo/43?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#39** <a href="https://app.graphite.dev/github/pr/sutrolabs/census-embedded-demo/39?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yeldah3 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: components/SyncObject.js [hunk 1]
```diff
@@ -242,6 +242,19 @@ export function SyncObject({
             )}
           </pre>
         }
+        body={
```
Comment (yeldah3) [inline components/SyncObject.js:245]: Forgot to add this request body on last PR!

File: components/Sidebar.js [hunk 2]
```diff
@@ -39,7 +42,13 @@ export default function Sidebar({ onLogOut }) {
...
         </nav>
 
-        <SidebarFooter onLogOut={onLogOut} />
+        <SidebarFooter
+          onLogOut={onLogOut}
+          embedMode={embedMode}
+          setEmbedMode={setEmbedMode}
+          devMode={devMode}
+          setDevMode={setDevMode}
+        />
```
Comment (LemonFace0309) [inline components/Sidebar.js:51]: this is great!

File: components/Sidebar.js [hunk 3]
```diff
@@ -63,29 +72,51 @@ function Item({ name, nested, href }) {
...
-      <p className="hidden md:block">Log out</p>
-    </button>
-    <div className="hidden items-center gap-4 text-2xl md:flex">
-      <Link
-        className="flex items-center gap-2 text-sm md:gap-3 md:text-2xl"
-        href="https://developers.getcensus.com/getting-started/introduction"
-      >
-        <a target="_blank">
-          <i className="fa-solid fa-book" />
+const SidebarFooter = ({ onLogOut, embedMode, setEmbedMode, devMode, setDevMode }) => (
```
Comment (LemonFace0309) [inline components/Sidebar.js:75]: we can probably rename this to `Sidebar` actually

File: components/Sidebar.js [hunk 4]
```diff
@@ -63,29 +72,51 @@ function Item({ name, nested, href }) {
...
-        href="https://github.com/sutrolabs/census-embedded-demo"
-      >
-        <a target="_blank">
-          <i className="fa-brands fa-github" />
+        <Tooltip anchorSelect="#embed-mode">
+          Show Census UI flows embedded directly in the application, or redirect to the flows.
+        </Tooltip>
+      </div>
+      <div className="flex items-center">
+        <span className="px-2">Dev Mode</span>
```
Comment (LemonFace0309) [inline components/Sidebar.js:88]: I think this messes up mobile a bit :(

CleanShot 2024-05-24 at 17 29 50@2x

File: components/Source.js [hunk 5]
```diff
@@ -201,20 +201,23 @@ export default function Source({
...
+              syncs={syncs.filter((sync) => sync.source_attributes.connection_id === source.id)}
               setSyncs={setSyncs}
               runsLoading={runsLoading}
               runs={runs}
               devMode={devMode}
-              embedSourceFlow={embedSourceFlow}
+              embedMode={embedMode}
+              addNewSyncText={"Configure data to import to Marketing Magnet"}
+              stepText={"Step 2: Choose which source objects to sync."}
+              useCase={"import"}
```
Comment (LemonFace0309) [inline components/Source.js:212]: this is great!

File: pages/integrations/export-ads.js [hunk 6]
```diff
@@ -13,7 +12,29 @@ export default function Index({
...
+  }
+
+  const isGoogleAudienceSync = (sync) => {
+    return (
+      destinationForSync(sync).type === "google_ads" && sync.destination_attributes.object === "user_data"
+    )
+  }
+
+  const googleAudienceSyncs = syncs.filter(isGoogleAudienceSync)
+  const facebookAudienceSyncs = syncs.filter(isFacebooksAudienceSync)
```
Comment (LemonFace0309) [inline pages/integrations/export-ads.js:36]: nicce. Nit but we can probably do a useMemo wrapper to memoize the outputs too

**Review by LemonFace0309 — COMMENTED** [review]
Looking really DRY!

**Pull request merged**